examples/cog-pyramid.js

```diff
@@ -0,0 +1,53 @@
+import GeoTIFF from '../src/ol/source/GeoTIFF.js';
+import LayerGroup from '../src/ol/layer/Group.js';
+import Map from '../src/ol/Map.js';
+import TileGrid from '../src/ol/tilegrid/TileGrid.js';
+import View from '../src/ol/View.js';
+import WebGLTileLayer from '../src/ol/layer/WebGLTile.js';
+
+// Metadata from https://s2downloads.eox.at/demo/EOxCloudless/2019/rgb/2019_EOxCloudless_rgb.json
+
+// Tile grid of the GeoTIFF pyramid layout
+const tileGrid = new TileGrid({
+  origin: [-180, 90],
+  resolutions: [0.703125, 0.3515625, 0.17578125, 8.7890625e-2, 4.39453125e-2],
+  tileSizes: [
+    [512, 256],
+    [1024, 512],
+    [2048, 1024],
+    [4096, 2048],
+    [4096, 4096],
+  ],
+});
+// Add a separate layer for each pyramid tile to a layer group
+const pyramid = new LayerGroup();
```

Comment:
I modified the example a bit to create pyramid tile layers on demand, only when a pyramid tile is used by the current map view. So now this will also work with GeoTIFF pyramids that have an arbitrary number of zoom levels and/or tiles per zoom level.

Reply:
Thanks a lot! This looks better in my opinion. For a production code I would use something like an LRU cache, so that we don't get swamped in layers, but for this example it is fine I guess.